### PR TITLE
[language/changelog] Add changelog entry for Move unit tests

### PR DESCRIPTION
**Pull request by tzakian**
Adds a language changelog entry for unit testing in Move

### CODE REVIEW

**Review by sblackshear — APPROVED** [review]

File: language/changes/4-unit-testing.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+
+```
+#[test] // OK
+fun this_is_a_test() { ... }
+
+#[test] // Will fail to compile since the test takes an argument
+fun this_is_not_correct(arg: signer) { ... }
+```
+
+A test can also be annotated as an `#[expected_failure]`. This annotation marks that the annotated test should abort -- however the exact abort code is not checked. The test will be marked as failing only if the test fails with a non-abort error, or does not abort. Only function that also have the `#[test]` annotation can also be annotated as an  #`[expected_failure]`.
```
Comment (bob-wilson) [inline language/changes/4-unit-testing.md:42]:
```suggestion
A test can also be annotated as an `#[expected_failure]`. This annotation marks that the annotated test should abort — however the exact abort code is not checked. The test will be marked as failing only if the test fails with a non-abort error, or does not abort. Only functions that have the `#[test]` annotation can also be annotated as an  #`[expected_failure]`.
```

File: language/changes/4-unit-testing.md [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+
+With arguments, a test annotation takes the form `#[test(<param_name_1> = <address>, ..., <param_name_n> = <address>)]`. If a function is annotated in such a manner, the function's parameters must be a permutation of the parameters <`param_name_1>, ..., <param_name_n>`, i.e., the order of these parameters as they occur in the function and their order in the test annotation do  not have to be the same, but they must be able to be matched up with each other by name.
+
+Only parameters with a type of `signer` are supported as test parameters. If a non-`signer` parameter is supplied, the test will result in an error when run.
+
+```
+#[test(arg = @0xC0FFEE)] // OK
+fun this_is_correct_now(arg: signer) { ... }
+
+#[test(wrong_arg_name = @0xC0FFEE)] // Not correct: arg name doesn't match`
```
Comment (bob-wilson) [inline language/changes/4-unit-testing.md:61]:
```suggestion
#[test(wrong_arg_name = @0xC0FFEE)] // Not correct: arg name doesn't match
```

File: language/changes/4-unit-testing.md [hunk 3]
```diff
@@ -0,0 +1,345 @@
+# Unit Testing
+
+* Status: Implemented in Move 1.3
+
+## Introduction
+
+Unit testing is a new feature in Move version 1.3 to allow Move programmers to more easily and concisely test individual functions and features. Previously, the only way to test Move code was using more heavyweight expected-value and transaction-based tests via the Move CLI and Diem functional test framework. Additionally, it was hard — if not impossible — to set up arbitrary state for testing, test non-public functions, and have test-only dependencies, functions, and datatypes. With the new Move unit testing feature these are all possible and easy to express.
```
Comment (bob-wilson) [inline language/changes/4-unit-testing.md:7]:
```suggestion
Unit testing is a new feature in Move version 1.3 to allow Move programmers to more easily and concisely test individual functions and features. Previously, the only way to test Move code was using more heavyweight expected-value and transaction-based tests via the Move CLI and Diem functional test framework. Additionally, it was hard — if not impossible — to set up arbitrary state for testing, to test non-public functions, and to have test-only dependencies, functions, and datatypes. With the new Move unit testing feature these are all possible and easy to express.
```

**Review by bob-wilson — APPROVED** [review]
I found a few more nits and suggested some changes. Looks good overall

**Review by bob-wilson — DISMISSED** [review]

**Pull request merged**